doc/import_export.rst

```diff
@@ -123,117 +88,62 @@ and their columns are a subset of names.
 	   capital_cost=1e5,
 	   p_max_pu=wind)
 
-Any attributes which are not specified will be given the default value from :doc:`components`.
-
 
 Adding components using pandas DataFrames
 =========================================
 
 To add multiple components whose static attributes are given in a
 pandas DataFrame, use
 
-``network.import_components_from_dataframe(dataframe, cls_name)``
-
-``dataframe`` is a pandas DataFrame whose index is the names of the
-components and whose columns are the non-default
-attributes. ``cls_name`` is the component name,
-e.g. ``"Line","Bus","Generator","StorageUnit``. If columns are missing
-then defaults are used. If extra columns are added, these are left in
-the resulting component DataFrame.
-
-.. code:: python
-
-    import pandas as pd
-
-    buses = ['Berlin', 'Frankfurt', 'Munich', 'Hamburg']
-
-    network.import_components_from_dataframe(pd.DataFrame({"v_nom" : 380,
-                                                           "control" : 'PV'},
-							  index=buses),
-					     "Bus")
-
-    network.import_components_from_dataframe(pd.DataFrame({"carrier" : "solar",
-                                                           "bus" : buses,
-							   "p_nom_extendable" : True,
-							   "capital_cost" : 6e4},
-							  index=[b+" PV" for b in buses]),
-					     "Generator")
+.. autofunction:: pypsa.io.import_components_from_dataframe
```

Comment:
The example could move into the docstring.

The docstring should have a note that there is `madd` now.

Reply:
Maybe `automethod`